drivers/pwm/pwm_nxp_flexio.c

```diff
@@ -0,0 +1,374 @@
+
+		} else {
+			/* Timer output is logic zero and is not affected by timer reset */
+			timerConfig.timerOutput = kFLEXIO_TimerOutputZeroNotAffectedByReset;
+			/* Set the timer mode to dual 8-bit counter PWM low */
+			timerConfig.timerMode = kFLEXIO_TimerModeDual8BitPWMLow;
+		}
+
+		/* Timer scaling factor w.r.t Flexio Clock */
+		timerConfig.timerDecrement = pwm_info->prescaler;
+
+		/* Program the PWM pulse */
+		timerConfig.timerCompare = ((uint8_t)(pwm_info->duty_cycle - 1U)) |
+			((uint8_t)(pwm_info->period - pwm_info->duty_cycle - 1U)
+			 << FLEXIO_PWM_TIMCMP_CMP_UPPER_SHIFT);
+
+		/* Configure Timer CFG and CTL bits to support PWM mode */
+		timerConfig.timerStop = kFLEXIO_TimerStopBitDisabled;
+		timerConfig.timerEnable = kFLEXIO_TimerEnabledAlways;
+		timerConfig.timerDisable = kFLEXIO_TimerDisableNever;
+		timerConfig.timerStart = kFLEXIO_TimerStartBitDisabled;
+		timerConfig.timerReset = kFLEXIO_TimerResetNever;
+		timerConfig.triggerSource = kFLEXIO_TimerTriggerSourceInternal;
+
+		/* Enable the pin out for the selected timer */
+		timerConfig.pinConfig = FLEXIO_PWM_TIMER_PIN_OUTPUT_ENABLE;
+		timerConfig.pinPolarity = pwm_info->polarity;
+
+		/* Select the pin that the selected timer will output the signal on */
+		timerConfig.pinSelect = pwm_info->pin_id;
+
+		FLEXIO_SetTimerConfig(flexio_base, child->res.timer_index[ch_id], &timerConfig);
+	}
+
+	return 0;
+}
+
+static const struct pwm_driver_api pwm_nxp_flexio_driver_api = {
+	.set_cycles = pwm_nxp_flexio_set_cycles,
+	.get_cycles_per_sec = pwm_nxp_flexio_get_cycles_per_sec,
+};
+
+#define _FLEXIO_PWM_PULSE_GEN_CONFIG(n)								\
+	{											\
+		.pin_id = DT_PROP(n, pin_id),							\
+		.prescaler = DT_PROP(n, prescaler),						\
+		.period = DT_PROP(n, period),							\
+		.duty_cycle = DT_PROP(n, duty_cycle),						\
+		.polarity = DT_PROP(n, polarity),						\
+	},
+
+#define FLEXIO_PWM_PULSE_GEN_CONFIG(n)								\
+	pwm_nxp_flexio_channel_config_type flexio_pwm_##n##_init[] = {				\
+		DT_INST_FOREACH_CHILD_STATUS_OKAY(n, _FLEXIO_PWM_PULSE_GEN_CONFIG)		\
+	};											\
+	const struct pwm_nxp_flexio_pulse_info flexio_pwm_##n##_info = {			\
+		.pwm_pulse_channels = ARRAY_SIZE(flexio_pwm_##n##_init),			\
+		.pwm_info = (pwm_nxp_flexio_channel_config_type *)flexio_pwm_##n##_init,	\
+	};
+
+#define FLEXIO_PWM_TIMER_INDEX_INIT(n)								\
+	uint8_t flexio_pwm_##n##_timer_index[ARRAY_SIZE(flexio_pwm_##n##_init)];
+
+#define FLEXIO_PWM_CHILD_CONFIG(n)								\
+	static const struct nxp_flexio_child mcux_flexio_pwm_child_##n = {			\
+		.isr = NULL,									\
+		.user_data = NULL,								\
+		.res = {									\
+			.shifter_index = NULL,							\
+			.shifter_count = 0,							\
+			.timer_index = (uint8_t *)flexio_pwm_##n##_timer_index,			\
+			.timer_count = ARRAY_SIZE(flexio_pwm_##n##_init)			\
+		}										\
+	};
+
+#define FLEXIO_PWM_PULSE_GEN_GET_CONFIG(n)							\
+	.pulse_info = (struct pwm_nxp_flexio_pulse_info *)&flexio_pwm_##n##_info,
```

Comment:
Instead of this typecasting you should rather define the `pulse_info` field as `const pwm_nxp_flexio_channel_config_type *`.

Reply:
pwm_nxp_flexio_pulse_info has 2 fields .pwm_pulse_channels (which is the number of pwm channels) and 
.pwm_info which is of type pwm_nxp_flexio_channel_config_type is an array with channel configurations for each channel.

Why do you want me to pulse_info to type channel_config ?

Reply:
I have removed the typecasting and made pulse_info field as const.